src/common/vector/value_vector_utils.cpp

```diff
@@ -7,19 +7,35 @@ using namespace common;
 
 void ValueVectorUtils::copyNonNullDataWithSameTypeIntoPos(
     ValueVector& resultVector, uint64_t pos, const uint8_t* srcData) {
-    copyNonNullDataWithSameType(resultVector.dataType, srcData,
-        resultVector.getData() + pos * resultVector.getNumBytesPerValue(),
-        resultVector.getOverflowBuffer());
+    if (resultVector.dataType.typeID == STRUCT) {
```

Comment:
now that you are changing this. I think we should discuss how LIST can be done in a similar fashion. ideally this copyInto and copyFrom should be handled with

```
switch typeID
recursive types:
   ...
non recursive types:
   ...
```
@ray6080 should think about this as well.